cmd/ztest/ztest.c

```diff
@@ -2660,6 +2662,66 @@ ztest_spa_create_destroy(ztest_ds_t *zd, uint64_t id)
 	(void) rw_unlock(&ztest_name_lock);
 }
 
+/*
+ * Start and then stop the MMP threads to ensure the startup and shutdown code
+ * works properly.  Actual protection and property-related code tested via ZTS.
+ */
+/* ARGSUSED */
+void
+ztest_mmp_enable_disable(ztest_ds_t *zd, uint64_t id)
+{
+	ztest_shared_opts_t *zo = &ztest_opts;
+	spa_t *spa = NULL;
+
+	if (zo->zo_mmp_test)
+		return;
+
+	mutex_enter(&spa_namespace_lock);
```

Comment:
Normally yes, but the way ztest is written there actually isn't any need to protect against the pool being removed from the namespace.  It's opened once in `ztest_run()` and then the test functions can safely assume `ztest_spa != NULL`.

The `ztest_name_lock` is used to protect against vdev addition/removal.  For this test case we want starting/stopping mmp to race with those vdev operations to verify it's safe so we shouldn't take that lock.

With the above in mind I'm going to remove `spa_namespace_lock` locking calls, and the `spa_open_ref` and `spa_close` calls to simplify this test function.  I'll update this PR with the change after an hour or so of local run time to verify there are no problems.